roles/openshift_facts/library/openshift_facts.py

```diff
@@ -519,83 +491,6 @@ def set_aggregate_facts(facts):
     return facts
 
 
-def set_deployment_facts_if_unset(facts):
-    """ Set Facts that vary based on deployment_type. This currently
-        includes master.registry_url
-
-        Args:
-            facts (dict): existing facts
-        Returns:
-            dict: the facts dict updated with the generated deployment_type
-            facts
-    """
-    if 'master' in facts:
-        deployment_type = facts['common']['deployment_type']
-        openshift_features = ['Builder', 'S2IBuilder', 'WebConsole']
-        if 'disabled_features' not in facts['master']:
-            if facts['common']['deployment_subtype'] == 'registry':
-                facts['master']['disabled_features'] = openshift_features
-        if 'registry_url' not in facts['master']:
-            registry_url = 'openshift/origin-${component}:${version}'
-            if deployment_type == 'openshift-enterprise':
-                registry_url = 'openshift3/ose-${component}:${version}'
-            facts['master']['registry_url'] = registry_url
-
-    return facts
-
-
-# pylint: disable=too-many-statements
-def set_version_facts_if_unset(facts):
-    """ Set version facts. This currently includes common.version and
-        common.version_gte_3_x
-
-        Args:
-            facts (dict): existing facts
-        Returns:
-            dict: the facts dict updated with version facts.
-    """
-    if 'common' in facts:
-        openshift_version = get_openshift_version(facts)
-        if openshift_version and openshift_version != "latest":
-            version = LooseVersion(openshift_version)
-            facts['common']['version'] = openshift_version
-            facts['common']['short_version'] = '.'.join([str(x) for x in version.version[0:2]])
```

Comment:
Some places still use `short_version`:

```
TASK [Update registry-console image to current version] ***************************************************************************************************************************************
task path: /home/vrutkovs/workdir/tmp.3zoyyI3KYA/openshift-ansible/playbooks/common/openshift-cluster/upgrades/post_control_plane.yml:84
Tuesday 06 March 2018  14:22:44 +0100 (0:00:01.610)       0:19:33.476 ********* 
fatal: [vrutkovs_tmp.3zoyyI3KYA-master-1]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: {{ openshift.master.registry_url | regex_replace ( '(origin|ose)-\\${component}', 'registry-console') | replace ( '${version}', 'v' ~ openshift.common.short_version ) }}: 'dict object' has no attribute 'short_version'\n\nThe error appears to have been in '/home/vrutkovs/workdir/tmp.3zoyyI3KYA/openshift-ansible/playbooks/common/openshift-cluster/upgrades/post_control_plane.yml': line 84, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: Update registry-console image to current version\n    ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: {{ openshift.master.registry_url | regex_replace ( '(origin|ose)-\\${component}', 'registry-console') | replace ( '${version}', 'v' ~ openshift.common.short_version ) }}: 'dict object' has no attribute 'short_version'"
}
```

Reply:
https://github.com/vrutkovs/openshift-ansible/commit/a594c3ba1b3ba4f4ec8e162aace86781cfd3f0d7 fixes it

Reply:
short_version will be removed from openshift_hosted in my other patch, which should merge before this one.  Removing it here will just make me rebase ;)

Reply:
Right, it works with all PRs applied